### PR TITLE
Record transfer limit for data refresh

### LINKED ISSUE #472: Capture subset of data for refresh on staging
## Problem

If we would like to run the data refresh in development, we currently don't have a way of capturing a subset of the data present in the catalog to ingest into the API.

## Description

It would be great if we had some method of gathering a semi-randomized subset of data when we know that the instance is running on staging. Something that would only transfer 100k rows from the catalog to the dev API database.

## Additional context

One way to accomplish this would be by adding an `ORDER BY identifier` and then `LIMIT 100000` in the initial select from the catalog. Since the GUIDs should be unique and non-sequential, forcing them to be sequential might allow us to easily and quickly simulate a random subset (since the `identifier` column is indexed on the catalog).

## Implementation

- [x] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #472 by @AetherUnbound

## Description

This PR adds a new variable, `DATA_REFRESH_LIMIT`, and some additional logic for interpreting an appropriate limit based on the environment the ingestion server is deployed in. In `prod` or `production`, the default limit is removed and all records are copied. In any other environment, at most 100k records are copied by default (although more or fewer can be specified using the aforementioned environment variable).

I also split the unit tests up and added tests for the altered query. This was a huge pain, because trying to convert the `psycopg2.SQL` objects into strings requires a database connection which is way too much overhead for such a simple test.

## Testing Instructions

`just ing-testlocal`

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: ingestion_server/ingestion_server/ingest.py [hunk 1]
```diff
@@ -298,7 +306,15 @@ def reload_upstream(table, progress=None, finish_time=None, approach="advanced")
 
         # Step 3: Import data into a temporary table
         log.info("Copying upstream data...")
-        copy_data = get_copy_data_query(table, shared_cols, approach=approach)
+        environment = config("ENVIRONMENT", default="local").lower()
+        limit_default = 100_000
+        if environment in {"prod", "production"}:
```
Comment (dhruvkb) [inline ingestion_server/ingestion_server/ingest.py:311]: We should settle on one of these in the future.

**Review by dhruvkb — APPROVED** [review]
LGTM, except a note for the future.

File: ingestion_server/ingestion_server/ingest.py (hunk 1 above)
Comment (AetherUnbound) [reply ingestion_server/ingestion_server/ingest.py:311]: Agreed, just trying to cover all my bases :sweat_smile: We've had some discussion about `local` as well in #475, I'll make an issue for unifying this everywhere.

**Review by zackkrida — APPROVED** [review]
Again, not the biggest 🐍 python authority but the code makes sense and I don't see any issues. Nice!

**Comment by AetherUnbound** [comment]
So I ran into that issue where it hangs on the cleaning step _again_, and I was able to resolve it by removing the garbage collection steps :upside_down_face: I'll have to do a deep dive on that at some point

**Pull request merged**